tests/test_trainer.py

```diff
@@ -670,6 +671,19 @@ def test_num_train_epochs_in_training(self):
         train_output = trainer.train()
         self.assertEqual(train_output.global_step, int(self.n_epochs))
 
+        # early stopping stops training early
```

Comment:
Let's put this in a new test, just to be cleaner. It's great you added one!

Reply:
Fixed in https://github.com/huggingface/transformers/pull/8581/commits/7f43217e908576bad51b05fc02202246ca3dfd4b